From 2b92fc046f13b1d2f5f6637283722aa28f050cb2 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Tue, 10 Jun 2025 10:10:49 +0300
Subject: [PATCH] [JSW-808] jgs: add global mem partition data

We have memory partition to sp sram, fw sram,
dccm for various use cases.
sp sram shared between boot info, preboot etc.
dccm shared with bootrom persistent memory
section. Hence, we need global header across
all fw modules.

Change-Id: Ieeae2fa6b0c99d008db2ded527a8238fc0751b19
---

diff --git a/gpu/jgs/mem_partition.h b/gpu/jgs/mem_partition.h
new file mode 100644
index 0000000..a79ec9e
--- /dev/null
+++ b/gpu/jgs/mem_partition.h
@@ -0,0 +1,27 @@
+/* SPDX-License-Identifier: MIT
+ *
+ * Copyright 2025 Intel Corporation.
+ *
+ */
+
+#ifndef _JGS_MEM_PARTITION_H_
+#define _JGS_MEM_PARTITION_H_
+
+/* NOTE: Important: use this header across project to follow common memory layout */
+
+#define BOOT_INFO_BASE			IO_0_SP_SRAM_BASE
+#define BOOT_INFO_SIZE			SZ_32K
+
+#define PPBOOT_TEXT_BASE		IO_0_PSOC_FW_SRAM_BASE
+
+#define PREBOOT_TEXT_BASE		(IO_0_SP_SRAM_BASE + BOOT_INFO_SIZE)
+#define PREBOOT_TEXT_BASE_FULL		(IO_0_SP_SRAM_BASE_FULL + BOOT_INFO_SIZE)
+
+/*
+ * In order not to conflict with ARC BOOTROM's usage of DCCM,
+ * Let PPBoot start from the offset of 0x6000, for using DCCM
+ * for its own BSS and Stack.
+ */
+#define PPBOOT_DCCM_BASE		0x80006000
+
+#endif /* _JGS_MEM_PARTITION_H_ */
